syntax = "proto3";

package ApWatchI;
option optimize_for = LITE_RUNTIME;

message AccessPoint
{
    string SSID = 1;
    uint32 SNR = 2;
    uint32 channel = 3;
}

message NewAp
{
    AccessPoint ap = 1;
}

message RemovedAp
{
    AccessPoint ap = 1;
}

message ModifiedApParams
{
    enum Param
    {
        None = 0;
        SNR = 1;
        channnel = 2;
    }

    AccessPoint oldAp = 1;
    AccessPoint newAp = 2;

    repeated Param changedParams = 3;
}

message Msg
{
    oneof msg {
        NewAp newAp = 1;
        RemovedAp removedAp = 2;
        ModifiedApParams modifiedAp = 3;
    }
}
